react/utils/metrics.ts

```diff
@@ -0,0 +1,148 @@
+import axios from 'axios'
+
+const ANALYTICS_URL = 'https://rc.vtex.com/api/analytics/schemaless-events'
+
+const GRAPHQL_URL = (accountName: string, workspace?: string) => {
+  if (workspace) {
+    return `https://${workspace}--${accountName}.myvtex.com/_v/private/graphql/v1`
+  }
+
+  return `https://${accountName}.myvtex.com/_v/private/graphql/v1`
+}
+
+type Metric = {
+  name: 'b2b-suite-buyerorg-data'
+  kind: 'create-quote-ui-event'
+  description: 'Create Quotation Action - UI'
+  account: string
+}
+
+type QuoteFieldsMetric = {
+  cost_center_id: string
+  cost_center_name: string
+  buy_org_id: string
+  buy_org_name: string
+  member_id: string
+  member_email: string
+  role: string
+  creation_date: string
+  quote_id: string
+  quote_reference_name: string
+  send_to_sales_rep: boolean
+}
+
+export type SessionProfile = {
+  id: { value: string }
+  email: { value: string }
+}
+
+type SessionResponse = {
+  namespaces: {
+    profile: SessionProfile
+    account: {
+      accountName: { value: string }
+    }
+  }
+}
+
+type MetricsParam = {
+  quoteId: string
+  sessionResponse: SessionResponse
+  workspace: string
+  sendToSalesRep: boolean
+}
+
+type QuoteMetric = Metric & { fields: QuoteFieldsMetric }
+
+const fetchMetricsData = async (
+  accountName: string,
+  workspace: string,
+  quoteId: string,
+  userEmail: string
+) => {
+  const query = JSON.stringify({
+    query: `query GetMetricsData($id: String, $email: String!) {
+      getQuote(id: $id) @context(provider: "vtex.b2b-quotes-graphql") {
+        organization
+        organizationName
+        costCenterName
+        referenceName
+        creatorRole
+        creationDate
+      },
+      getUserByEmail(email: $email) @context(provider: "vtex.storefront-permissions") {
+      costId
+    }
+  }`,
+    variables: { id: quoteId, email: userEmail },
+  })
+
+  const { data } = (
+    await axios.post(GRAPHQL_URL(accountName, workspace), query)
+  ).data
+
+  const quoteResult = data.getQuote as Omit<QuoteMetricsData, 'costId'>
+  const costId = data.getUserByEmail?.[0].costId as string
```

Comment:
You're using an optional operator here, so you're assuming that `data.getUserByEmail` might not be set, right? If that's a possibility then you shouldn't cast to string here, as it'll make `costId`'s type not consider the use case where it will be `undefined`. Is it ok if `costId` isn't set? If it isn't it's better to bail early here, and if it's ok then you should let it be `string | undefined`.